truffle/src/main/java/org/truffleruby/core/module/ConstantLookupResult.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.core.module;
+
+import java.util.ArrayList;
+
+import org.truffleruby.RubyContext;
+import org.truffleruby.language.LexicalScope;
+import org.truffleruby.language.RubyConstant;
+
+import com.oracle.truffle.api.Assumption;
+import com.oracle.truffle.api.object.DynamicObject;
+
+public class ConstantLookupResult {
+
+    private final RubyConstant constant;
+    private final Assumption[] assumptions;
+
+    public ConstantLookupResult(RubyConstant constant, Assumption assumption) {
+        this.constant = constant;
+        this.assumptions = new Assumption[]{ assumption };
+    }
+
+    public ConstantLookupResult(RubyConstant constant, ArrayList<Assumption> assumptions) {
```

Comment:
Could we simplify the interface to this class and have just one version with an `Assumption...` parameter? If people want to use lists to build up their assumptions they can do but they should do the work of converting to an array, rather than just dictating `List`, let alone `ArrayList`.